src/services/challenges_repository.js

```diff
@@ -50,14 +61,19 @@ export default class ChallengesRepository {
     const resolvedChallengeEvents = await this.challengesWrapper.resolvedChallenges(fromBlock);
     const timedOutChallengeEvents = await this.challengesWrapper.timedOutChallenges(fromBlock);
 
-    return this.sortChallenges(
-      this.filterOutFinishedChallenges(
-        this.extractChallengeFromEvent(allChallengeEvents, ['challengeId', 'sheltererId', 'bundleId', 'count']),
-        this.extractChallengeFromEvent(resolvedChallengeEvents, ['challengeId', 'resolverId']),
-        this.extractChallengeFromEvent(timedOutChallengeEvents, ['challengeId']),
-        this.challengesWrapper.defaultAddress
-      )
+    const startedChallenges = this.extractChallengeFromEvent(allChallengeEvents, ['challengeId', 'sheltererId', 'bundleId', 'count']);
+    const resolvedChallenges = this.extractChallengeFromEvent(resolvedChallengeEvents, ['challengeId', 'resolverId']);
+    const timedOutChallenges = this.extractChallengeFromEvent(timedOutChallengeEvents, ['challengeId']);
+    const filtered = this.filterOutFinishedChallenges(
+      startedChallenges,
+      resolvedChallenges,
+      timedOutChallenges,
+      this.challengesWrapper.defaultAddress
+    );
+    const sorted = this.sortChallenges(
+      filtered
     );
+    return sorted;
```

Comment:
This is a little bit artificial (we could just do a straight return), but makes the code more readable, and adding performance measure points easier.